routes/projects/files/create-update.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile
+ * The :pathToFile parameter must be URI encoded
+ * Create or replace the contents for the given file path
+ * The message body that would normally be expected by this route
+ * will look something like:
+ * {
+ *   type: <string> content-type,
+ *   path: <string>,
+ *   data: <string> base64 encoded binary data
```

Comment:
I'd suggest we always send/receive `ArrayBuffer` on the browser side, and always send/receive `Buffer` on the server side.  We don't need to encode at all.